clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -513,8 +515,11 @@ private ClientResponse sendAndAwaitInitProducerIdRequest(Node node) throws IOExc
         return NetworkClientUtils.sendAndReceive(client, request, time);
     }
 
-    private Node awaitLeastLoadedNodeReady(long remainingTimeMs) throws IOException {
-        Node node = client.leastLoadedNode(time.milliseconds());
+    private Node awaitLeastLoadedNodeReady(long remainingTimeMs, FindCoordinatorRequest.CoordinatorType coordinatorType) throws IOException {
```

Comment:
nit: the name should be changed since we do not always use the least loaded node. 

I noticed we are always using request timeout for the argument. Maybe we can drop the argument?